kani-compiler/src/codegen_cprover_gotoc/codegen/intrinsic.rs

```diff
@@ -239,6 +239,19 @@ impl<'tcx> GotocCtx<'tcx> {
         // *var1 = op(*var1, var2);
         // var = tmp;
         // -------------------------
+        //
+        // In fetch functions of atomic_ptr, the type of var2 can be pointer (invalid_mut).
+        // In such case, atomic binops are transformed as follows to avoid typecheck failure.
+        //
+        // -------------------------
+        // var = atomic_op(var1, var2)
+        // -------------------------
+        // unsigned char tmp;
+        // tmp = *var1;
+        // *var1 = (typeof var1)op((size_t)*var1, (size_t)var2);
```

Comment:
This is the only line that changes w.r.t. the previous example. Why don't we edit the original example so that the possibility of `var2` being a pointer is considered? Something like:
```
        // unsigned char tmp;
        // tmp = *var1;
        // if var2.is_pointer() {
        //   *var1 = (typeof var1)op((size_t)*var1, (size_t)var2);
        // } else {
        //   *var1 = op(*var1, var2);
        // }
        // var = tmp;
```

Reply:
The two examples are the two cases of how we transform atomic binop into goto. There is no if-then-else in the transformed snippet. Combining the two examples and introducing an if-then-else statement in it will be misleading.